lib/cdo/aws/s3.rb

```diff
@@ -90,6 +90,35 @@ def self.exists_in_bucket(bucket, key)
       return false
     end
 
+    # Returns true iff the specified S3 key exists in the buckep
+    #
+    # Will query all objects in the bucket up front and cache that result. Note
+    # this cache will not expire, so this method is not recommended for use in
+    # production. The create and delete operations defined in this module do
+    # attempt to keep this cache up-to-date, but other operations could easily
+    # modify the bucket contents and render this method unreliable.
+    #
+    # Note also that we are not using the Rails cache framework for this. We
+    # rely on this method for the text-to-speech update; specifically the
+    # translation-aware tts update we do as part of the i18n sync, which
```

Comment:
We also call this method indirectly from https://github.com/code-dot-org/code-dot-org/blob/eae96967671ec46faa85068b74e7493c5573c362/dashboard/app/models/script.rb#L784, which I don't think is i18n-only (not sure though). Will this caching cause any problems here?

Reply:
That's a convenience method, intended to make it easier for folks to manually update tts. We don't actually use it anywhere!